stumpy/stamp.py

```diff
@@ -126,15 +130,30 @@ def stamp(T_A, T_B, m, ignore_trivial=False):
     T_B.shape[0]-m+1
     """
 
+    # Preprocessing to remove nan and inf values
+    if T_A.ndim != 1:  # pragma: no cover
+        raise ValueError(f"T_A is {T_A.ndim}-dimensional and must be 1-dimensional. ")
+
+    T_A = T_A.copy()
+    T_A[np.isinf(T_A)] = np.nan
     core.check_dtype(T_A)
-    core.check_nan(T_A)
+
+    T_B = T_B.copy()
+    T_B[np.isinf(T_B)] = np.nan
+    if T_B.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T_B is {T_B.ndim}-dimensional and must be 1-dimensional. "
+            "For multidimensional STUMP use `stumpy.mstump` or `stumpy.mstumped`"
+        )
     core.check_dtype(T_B)
-    core.check_nan(T_B)
+
     core.check_window_size(m)
     subseq_T_B = core.rolling_window(T_B, m)
     excl_zone = int(np.ceil(m / 2))
     M_T, Σ_T = core.compute_mean_std(T_A, m)
 
+    T_A[np.isnan(T_A)] = 0
+
```

Comment:
2. is almost correct, but we convert to NaNs because of the `np.nanstd` function that ignores NaNs, as we do not want NaNs in our final Σ_T, because of warings raised by numpy when dividing by NaN in the distance profile calculation. But, I just saw this when I was trying to fix the Memory Error, `np.nanstd([np.nan, np.nan]) = np.nan`, so if T contains more NaNs in a row than the window size, this still happens. So at the end of compute_mean_std there should be something like `Σ_T[np.isnan(Σ_T)] = 0`
3. correct. We can set an arbitrary value there, because we do not care about these subsequences and the matrix profile is going to be set to `np.inf` regardless.
4. The distance will be `np.inf` since its mean will be 'np.inf` (see stumpy.core line 304). Its stddev will be 0, as noted in point three. There are two possibilities that can happen:
Say we want to calculate the distance profile of a subsequence Q to T.
1. If Q contains any nan value, then the distance profile will be `np.inf` everywhere. If the mean of Q is infinity, then Q contained a nan value *before* we replaced them all by 0 (Step 3).
2. If Q does not contain any nan value, then we calculate the distance profile. However, T can have nan values, so we might have to set the distance profile at some points to infinity (for all sub-sequences in T that contain a nan value, i.e whose mean is np.inf). 

Both cases are covered in core line 352/359, in the calculation of the distance profile, because `D_squared[i] = np.inf` if and only if `M_T[i] = np.inf` or `μ_Q = np.inf`. So we maybe should remove line 141 from stomp (and similar lines in stump and stumped).